src/main/java/net/minecraftforge/client/event/RegisterPresetEditorsEvent.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.client.event;
+
+import java.util.Map;
+
+import org.jetbrains.annotations.ApiStatus;
+
+import net.minecraft.client.gui.screens.worldselection.PresetEditor;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.world.level.levelgen.presets.WorldPreset;
+import net.minecraftforge.eventbus.api.Cancelable;
+import net.minecraftforge.eventbus.api.Event;
+import net.minecraftforge.fml.LogicalSide;
+import net.minecraftforge.fml.event.IModBusEvent;
+import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;
+
+/**
+ * <p>Event for registering {@link PresetEditor} screen factories for world presets.</p>
+ *
+ * <p>This event is not {@linkplain Cancelable cancellable}, and does not {@linkplain HasResult have a result}.
+ *
+ * <p>This event is fired on the {@linkplain FMLJavaModLoadingContext#getModEventBus() mod-specific event bus},
+ * only on the {@linkplain LogicalSide#CLIENT logical client}.</p>
+ */
+public class RegisterPresetEditorsEvent extends Event implements IModBusEvent
+{
+    private final Map<ResourceKey<WorldPreset>, PresetEditor> editors;
```

Comment:
What is the reason why this just uses resource keys whereas mojang uses optionals of resource keys? I agree that resource keys seem to make sense but I can't figure out what the reason that mojang might have for using optionals in the first place

Reply:
I honestly don't know why mojang uses optionals there, I can only assume it's because the registry/holder API provides optional keys and they didn't feel like using map/ifpresent. The vanilla map has three non-empty optionals in its keys and nothing else.

Reply:
Might be worth a brief comment that points out or at least clarifies that it is intended this is not using optionals even though vanilla does just so that during ports it doesn't potentially get thought of as a porting desync